monkey/infection_monkey/utils/threading.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from threading import Event, Thread
+from typing import Any, Callable, Iterable, Tuple
+
+logger = logging.getLogger(__name__)
+
+
+def run_worker_threads(target: Callable[..., None], args: Tuple = (), num_workers: int = 2):
+    worker_threads = []
+    for i in range(0, num_workers):
+        t = create_daemon_thread(target=target, args=args)
+        t.start()
+        worker_threads.append(t)
+
+    for t in worker_threads:
+        t.join()
+
+
+def create_daemon_thread(target: Callable[..., None], args: Tuple = ()):
+    return Thread(target=target, args=args, daemon=True)
+
+
+def interruptable_iter(
+    iterator: Iterable, interrupt: Event, log_message: str = None, log_level: int = logging.DEBUG
```

Comment:
There's still a bit of duplication in that all users of this method initialize essentially the same stop event and all pass the same stop event here. Maybe we could have a `StoppableThread` class or something. Also, logging sounds like a different responsibility. But good enough for now

Reply:
We can improve on it as we go. I know what you mean about logging, but I'm not sure of a better way to do it at this time.

The fact that everything is using the same stop event is incidental. For example, it cannot be assumed that the stop event in the `Propagator` and `Ransomware` payload are the same. For the moment they are, but we may want to be able to stop these activities independently in the future.